src/components/Image/index.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import {Image as RNImage} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import PropTypes from 'prop-types';
+import ONYXKEYS from '../../ONYXKEYS';
+import {defaultProps, imagePropTypes} from './imagePropTypes';
+import RESIZE_MODES from './resizeModes';
+
+const propTypes = {
+    ...imagePropTypes,
+
+    /** The URI source of the image */
+    source: PropTypes.shape({
+        uri: PropTypes.string.isRequired,
+    }).isRequired,
+};
+
+class Image extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            imageSource: undefined,
+        };
+    }
+
+    componentDidMount() {
+        this.configureImageSource();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.source === this.props.source) {
+            return;
+        }
+        this.configureImageSource();
+    }
+
+    // Check if the image source is a URL - if so the `encryptedAuthToken` is appended
+    // to the source. The natural image dimensions can then be retrieved using this source
+    // and as a result the `onLoad` event needs to be maunually invoked to return these dimensions
+    configureImageSource() {
```

Comment:
This should also be a "method doc" type of a comment

```suggestion
    /**
     * Check if the image source is a URL - if so the `encryptedAuthToken` is appended
     * to the source. The natural image dimensions can then be retrieved using this source
     * and as a result the `onLoad` event needs to be maunually invoked to return these dimensions
     */
    configureImageSource() {
```